app/scripts/shell/dimAppCtrl.controller.js

```diff
@@ -162,27 +162,35 @@
       }
     };
 
+    // Don't refresh more than once a minute
+    var refresh = _.throttle(vm.refresh, 60 * 1000);
+
     vm.startAutoRefreshTimer = function () {
       var secondsToWait = 360;
 
       $rootScope.autoRefreshTimer = $interval(function () {
        //Only Refresh If We're Not Already Doing Something
        //And We're Not Inactive
-       if (!$rootScope.loadingTracker.active() && !$rootScope.isUserInactive()) {
-         vm.refresh();
+       if (!$rootScope.loadingTracker.active() && !$rootScope.isUserInactive() && document.visibilityState == 'visible') {
+         refresh();
        }
       }, secondsToWait * 1000);
     };
 
     vm.startAutoRefreshTimer();
 
-    var refresh = _.debounce(vm.refresh, 300);
-
     $rootScope.$on('dim-settings-updated', function(event, arg) {
       // if ((!_.has(arg, 'charCol')) && (!_.has(arg, 'vaultCol'))) {
         refresh();
       // }
     });
+
+    // Refresh when the user comes back to the page
+    document.addEventListener("visibilitychange", function() {
```

Comment:
If I can make everything fast, can we keep this behavior. It feels great to have everything just update automatically :-)